src/main/resources/org/springframework/integration/config/xml/spring-integration-kafka-1.2.xsd

```diff
@@ -105,6 +105,30 @@
                                             ]]></xsd:documentation>
 										</xsd:annotation>
 									</xsd:attribute>
+									<xsd:attribute name="key-class-type" use="optional" type="xsd:string" default="[B">
+										<xsd:annotation>
+											<xsd:appinfo>
+												<xsd:documentation>
+													Class type used for the value
+												</xsd:documentation>
+												<tool:annotation kind="direct">
+													<tool:expected-type type="java.lang.Class"/>
+												</tool:annotation>
+											</xsd:appinfo>
+										</xsd:annotation>
+									</xsd:attribute>
+									<xsd:attribute name="value-class-type" use="optional" type="xsd:string" default="[B">
+										<xsd:annotation>
+											<xsd:appinfo>
+												<xsd:documentation>
+													Class type used for the value
+												</xsd:documentation>
+												<tool:annotation kind="direct">
+													<tool:expected-type type="java.lang.Class"/>
+												</tool:annotation>
+											</xsd:appinfo>
+										</xsd:annotation>
+									</xsd:attribute>
 									<xsd:attribute name="value-encoder" use="optional" type="xsd:string">
```

Comment:
Given that we have some major schema incompatibilities anyway, why do we need to continue to support Encoder?

Seems like we can still provide the adapter as a helper for users who have custom encoders but it's not unreasonable to have them do the wrapping themselves.